.github/workflows/pull-request.yml

```
@@ -4,137 +4,615 @@
+    runs-on: ubuntu-latest
+    needs: [Code-Quality-Checks]
     steps:
-      - uses: actions/checkout@v3
-      - name: 'Check whether comments exists for JSDocs'
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
```

actionlint (1.7.10)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)